### PR TITLE
Re-add grace period to Upload completer for backwards compatibility.

### PR DESCRIPTION
**Pull request by Joerger**
Fixes backwards compatibility issues that came from https://github.com/gravitational/teleport/pull/11551

Specifically, If a node starts a session recording in proxy/sync modes (which records the session directly on the Proxy/Auth servers), then the upload completer in the Proxy/Auth services will periodically check for a corresponding `SessionTracker` for any uncompleted uploads on disk.

Since pre-v9 sessions do not create session trackers for ongoing sessions, the Proxy/Auth servers will incorrectly assume that their uploads are abandoned. As a result, the upload will be completed prematurely and result in an error in the session due to the audit writer being closed on Proxy/Auth side.

Changes:
- Re-add grace period to Upload completer for backwards compatibility.
- Provide grace period to upload completer in Proxy and Auth services so that proxy/sync recording works with old nodes.
- Further decouple uploader from upload completer so that they can be configured as standalone processes more easily.

TODO: manually test and fix any failing tests

### CODE REVIEW

**Comment by ibeckermayer** [comment]
@zmb3 do we have a similar thing in `windows_desktop_service`? It seems all the other major feature services are changed here but ours.

**Comment by zmb3** [comment]
> @zmb3 do we have a similar thing in `windows_desktop_service`? It seems all the other major feature services are changed here but ours.

You know now that you say this, we might a long-standing bug that the upload completer will not run if you run a standalone desktop service!

File: lib/events/complete.go [hunk 1]
```diff
@@ -153,6 +165,16 @@ func (u *UploadCompleter) checkUploads(ctx context.Context) error {
 
 	// Complete upload for any uploads without an active session tracker
 	for _, upload := range uploads {
+		// DELETE IN 11.0.0
+		// To support v9/v8 versions which do not use SessionTrackerService,
+		// sessions are only considered abandoned after the provided grace period.
+		if u.cfg.GracePeriod != 0 {
+			gracePoint := upload.Initiated.Add(u.cfg.GracePeriod)
+			if gracePoint.After(u.cfg.Clock.Now()) {
+				return nil
```
Comment (ibeckermayer) [inline lib/events/complete.go:174]: Should this not be a `continue`?

**Review by ibeckermayer — COMMENTED** [review]
Straining to wrap my head around an unfamiliar part of the code and so could well be wrong, but may have found a logic bug in the change to `checkUploads`.

**Review by ibeckermayer — APPROVED** [review]

**Review by probakowski — APPROVED** [review]
LGTM

**Pull request merged**